compiler/rustc_lint/src/non_fmt_panic.rs

```diff
@@ -309,14 +309,21 @@ fn panic_call<'tcx>(cx: &LateContext<'tcx>, f: &'tcx hir::Expr<'tcx>) -> (Span,
     // Unwrap more levels of macro expansion, as panic_2015!()
     // was likely expanded from panic!() and possibly from
     // [debug_]assert!().
-    for &i in
-        &[sym::std_panic_macro, sym::core_panic_macro, sym::assert_macro, sym::debug_assert_macro]
-    {
+    loop {
```

Comment:
This...isn't exactly the same, right?

Previously, we can only match each diagnostic item once? But now we can match the same on continously?

Also, now we can ascend many times.

I'm not sure there are actually any semantic changes, just worth mentioning.

Reply:
In the current implementation, I think the order of the list is significant because `assert` contains a call to `panic`, and we want to climb to the outermost macro call. This implementation works differently but should have the same result AFAICT.